keysign/discover.py

```diff
@@ -28,13 +28,23 @@ def __init__(self, userdata, discovery):
             self.discovery = AvahiKeysignDiscoveryWithMac()
         self.bt = None
         self.stopped = False
-
+    
     @inlineCallbacks
     def start(self):
         # First we try Avahi, if it fails we fallback to Bluetooth because
         # the receiver may be able to use only one of them
         log.info("Trying to use this code with Avahi: %s", self.userdata)
-        key_data = yield threads.deferToThread(self.discovery.find_key, self.userdata)
+        
+        try:
+            key_data = yield threads.deferToThread(self.discovery.find_key, self.userdata)
+            log.debug("Received key successfully")
+        except Exception as e:
```

Comment:
Exactly catch the exception that is raised instead of all the possible exceptions